config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java

```diff
@@ -751,35 +751,42 @@ static File applicationFile(File parent, Path path) {
 
     /* Validates that files in application dir and subdirectories have a known extension */
     public void validateFileExtensions(boolean throwIfInvalid) {
-        validFileExtensions.forEach((key, value) -> {
-            java.nio.file.Path path = appDir.toPath().resolve((key.toFile().toPath()));
-            File dir = path.toFile();
-            if ( ! dir.exists() || ! dir.isDirectory()) return;
+        validFileExtensions.forEach((subDir, __) -> validateInDir(subDir.toFile().toPath(), throwIfInvalid));
+    }
 
-            try (var filesInPath = Files.list(path)) {
-                filesInPath.forEach(f -> validateFileSuffix(path, f, value, throwIfInvalid));
-            } catch (IOException e) {
-                log.log(Level.WARNING, "Unable to list files in " + dir, e);
-            }
-        });
+    private void validateInDir(java.nio.file.Path subDir, boolean throwIfInvalid) {
+        java.nio.file.Path path = appDir.toPath().resolve(subDir);
+        File subDirectory = path.toFile();
+        if ( ! subDirectory.exists() || ! subDirectory.isDirectory()) return;
+
+        try (var filesInPath = Files.list(path)) {
+            filesInPath.forEach(filePath -> {
+                if (filePath.toFile().isDirectory())
+                    validateInDir(appDir.toPath().relativize(filePath), throwIfInvalid);
+                else
+                    validateFileSuffix(filePath, throwIfInvalid);
+            });
+        } catch (IOException e) {
+            log.log(Level.WARNING, "Unable to list files in " + subDirectory, e);
+        }
     }
 
     static {
         validFileExtensions = Map.ofEntries(
                 Map.entry(Path.fromString(COMPONENT_DIR), Set.of(".jar")),
                 Map.entry(CONSTANTS_DIR, Set.of(".json", ".json.lz4")),
                 Map.entry(Path.fromString(DOCPROCCHAINS_DIR), Set.of(".xml")),
-                Map.entry(MODELS_DIR, Set.of(".model")),
+                // Map.entry(MODELS_DIR, Set.of(".model")),  TODO: Enable on Vespa 8
```

Comment:
Should add .onnx and .txt here at least. But since people can write custom models, like e.g embedders consuming files in any formats they like, and it could be logical to place these in models, perhaps we shouldn't validate this dir at all?

Reply:
Yes, I think we should skip validating files in models/ , there are too many (unknown) valid extensions.